cpp/src/arrow/compute/kernels/codegen_internal_test.cc

```diff
@@ -156,6 +156,24 @@ TEST(TestDispatchBest, CommonTemporal) {
   args = {timestamp(TimeUnit::SECOND, "America/Phoenix"),
           timestamp(TimeUnit::SECOND, "UTC")};
   ASSERT_EQ(nullptr, CommonTemporal(args.data(), args.size()));
+  args = {date32(), duration(TimeUnit::MILLI)};
+  AssertTypeEqual(duration(TimeUnit::MILLI), CommonTemporal(args.data(), args.size()));
+  args = {date64(), duration(TimeUnit::MICRO)};
+  AssertTypeEqual(duration(TimeUnit::MICRO), CommonTemporal(args.data(), args.size()));
```

Comment:
Check that timestamp + duration = timestamp?

Reply:
That falls more into [ARROW-14095](https://github.com/apache/arrow/pull/12137/files) and here I'm only doing the minimum to pass test cases. Do you think it makes sense to do the whole logic at once?

Reply:
Ah, ok, sounds good then. (I wonder if these JIRAs got split up a little too much, but oh well.)

Reply:
Yeah perhaps :).